lib/createContainer/createContainer.js

```diff
@@ -39,11 +39,14 @@ function createContainer(InnerComponent, config) {
         observer.dispose();
       }
     },
+    componentWillReceiveProps(nextProps) {
+      config.props = nextProps;
+    },
     getInitialState() {
       config = _.defaults(config, {
         // Have a default implementation of done so it can be
         // called from other handlers
-        done: (results) => {
+        done: function (results) {
```

Comment:
Its the initial component nonetheless. So if you have unmounted it and then mounted it THIS inside default done is the first now unmounted component. With the old props and such.